filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsPathBuilder.scala

```diff
@@ -266,3 +270,46 @@ case class GcsPath private[gcs] (nioPath: NioPath,
 
   private def userProjectBlobTarget: List[BlobTargetOption] = List(BlobTargetOption.userProject(projectId))
 }
+
+object GcsPath {
+
+  // When generating "identity" hashes for GCS files (see cromwell.core.callcaching.FileHashStrategy), we use the
+  // blob id, which includes bucket name, blob name, and generation. This slash-delimited string is md5-hashed to
+  // obtain a string of a consistent length. This is used for call caching. Logic for generating this fingerprint
+  // is centralized here because different codepaths are dealing with different representations of the GS object.
+
```

Comment:
Would love to DRY the below a little more, but not sure I can without finding the magical Google library type that `StorageObject` and `Blob` both inherit from. I've poked around a little for that and haven't found anything useful.